app/models/health/team_performance.rb

```diff
@@ -0,0 +1,292 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Health
+  class TeamPerformance
+    include ArelHelper
+
+    attr_accessor :range
+    def initialize range:, team_scope: nil
+      @range = (range.first.to_date..range.last.to_date)
+      @team_scope = team_scope
+    end
+
+    def self.url
+      'warehouse_reports/health/agency_performance'
+    end
+
+    def team_counts
+      @team_counts ||= teams.map do |id, name|
+        patient_ids = patient_referrals.select do |_, (_, team_id)|
+          team_id == id
+        end.keys
+
+        consented_patients = consent_dates.select { |p_id, _| p_id.in?(patient_ids) }.keys
+        unconsented_patients = patient_ids - consented_patients
+
+        with_ssms = ssm_dates.select { |p_id, _| p_id.in?(patient_ids) }.keys
```

Comment:
This really doesn't matter, but I wonder if these would be more readable like this:
```suggestion
        with_ssms = ssm_dates.keys & patient_ids
```

Reply:
the suggestion is roughly 10x faster, which probably doesn't matter at the scale of this report, but might be worth implementing anyway.